persistence/persistence/src/main/java/one/microstream/persistence/types/PersistenceStorerDeactivateAble.java

```diff
@@ -0,0 +1,182 @@
+package one.microstream.persistence.types;
+
+/*-
+ * #%L
+ * MicroStream Persistence
+ * %%
+ * Copyright (C) 2019 - 2022 MicroStream Software
+ * %%
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License, v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is
+ * available at https://www.gnu.org/software/classpath/license.html.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
+/**
+ * A {@link one.microstream.persistence.types.PersistenceStorer PersistenceStorer} implementation that allows
+ * switching between the supplied {@code PersistenceStorer} instance and a
+ * {@link one.microstream.persistence.types.PersistenceStorerDeactivated PersistenceStorerDeactivated}
+ * instance.
+ *
+ */
+public class PersistenceStorerDeactivateAble implements PersistenceStorer {
+
+	///////////////////////////////////////////////////////////////////////////
+	// instance fields //
+	////////////////////
+	
+	private       PersistenceStorer actual;
+	private final PersistenceStorer fullFeaturedStorer;
+	private final PersistenceStorer noOpStorer;
+	
+	
+	///////////////////////////////////////////////////////////////////////////
+	// constructors //
+	/////////////////
+	
+	public PersistenceStorerDeactivateAble(final PersistenceStorer persistenceStorer)
+	{
+		this.fullFeaturedStorer = persistenceStorer;
+		this.noOpStorer = new PersistenceStorerDeactivated();
+		this.actual = this.fullFeaturedStorer;
+	}
+	
+	
+	///////////////////////////////////////////////////////////////////////////
+	// methods //
+	////////////
+	
+	/**
+	 * Enable or disable writing support.
+	 * 
+	 * @param enabledWrites true to enable writing, false to disable.
+	 */
+	public void enableWrites(final boolean enabledWrites)
```

Comment:
We should go with getter/setter style naming, here setWriteEnabled(boolean), and also add a getter (isWriteEnabled)

Reply:
ok, done